nexus_constructor/transformations.py

```diff
@@ -157,14 +157,29 @@ def ui_value(self) -> float:
         Used for getting the 3d view magnitude (as a placeholder or if the dataset is scalar)
         :return:
         """
-        if isinstance(self.dataset, h5py.Dataset) and np.isscalar(self.dataset[()]):
-            try:
-                float(self._dataset[()])
-                return self._dataset[()]
-            except ValueError:
-                logging.debug(
-                    "transformation value is not cast-able to int, using UI placeholder value instead."
-                )
+        if isinstance(self.dataset, h5py.Dataset):
+            if np.isscalar(self.dataset[()]):
+                try:
+                    self.ui_value = float(self._dataset[()])
+                    return float(self._dataset[()])
+                except ValueError:
+                    logging.debug(
+                        "transformation value is not cast-able to float/int, using UI placeholder value instead."
+                    )
+            else:
+                # Dataset value is array - use the first value of the array as the UI value
+                try:
+                    self.ui_value = float(self._dataset[...][0])
+                    return float(self._dataset[...][0])
+                except ValueError:
+                    # not int-type
```

Comment:
comment incorrect, should be something like "not castable to float" maybe?
Rather than `pass` should it return a default value (0.0)?
Could have a `default_value` variable and use that also on lines 180 and 181.